ssl/ssl_lib.c

```diff
@@ -3084,6 +3084,14 @@ void ssl_set_masks(SSL *s)
 
     if (rsa_enc || rsa_sign) {
         mask_a |= SSL_aRSA;
+        /*
+         * If we only have an RSA-PSS certificate allow RSA authentication
+         * if TLS 1.2 and peer supports it.
+         */
+    } else if (ssl_has_cert(s, SSL_PKEY_RSA_PSS_SIGN)
```

Comment:
It might be more clear to make this a standalone 'if' clause instead of 'else if' (and outdent the comment to match, moving it outside the braces) -- there shouldn't be any functional change from doing so.